test/e2e/specs/onboarding/signup__free.ts

```diff
@@ -0,0 +1,118 @@
+/**
+ * @group calypso-release
+ */
+
+import {
+	DataHelper,
+	FreeSignupFlow,
+	SidebarComponent,
+	RestAPIClient,
+	UserSignupPage,
+} from '@automattic/calypso-e2e';
+import { Page, Browser } from 'playwright';
+import { apiCloseAccount } from '../shared';
+import type { NewUserResponse } from '@automattic/calypso-e2e';
+
+declare const browser: Browser;
+
+describe( 'Signup: Create a WordPress.com Free site as a new user', function () {
+	const testUser = DataHelper.getNewTestUser( {
+		usernamePrefix: 'signupfree',
+	} );
+
+	let page: Page;
+	let userDetails: NewUserResponse;
+	let userCreatedFlag = false;
+
+	beforeAll( async () => {
+		page = await browser.newPage();
+	} );
+
+	describe( 'Register as new user', function () {
+		it( 'Navigate to the WordPress.com Free signup page', async function () {
+			await page.goto( DataHelper.getCalypsoURL( 'setup/free' ) );
+		} );
+
+		it( 'Sign up as new user', async function () {
+			const userSignupPage = new UserSignupPage( page );
+			userDetails = await userSignupPage.signup(
+				testUser.email,
+				testUser.username,
+				testUser.password
+			);
+
+			userCreatedFlag = true;
+		} );
+	} );
+
+	describe( 'Onboarding', function () {
+		let freeSignupFlow: FreeSignupFlow;
+
+		beforeAll( async function () {
+			freeSignupFlow = new FreeSignupFlow( page );
+		} );
+
+		it( 'Enter site name and description', async function () {
+			await freeSignupFlow.enterSiteName( testUser.siteName );
+			await freeSignupFlow.enterDescription( DataHelper.getRandomPhrase() );
+
+			await Promise.all( [
+				page.waitForURL( /setup\/free\/designSetup/ ),
+				freeSignupFlow.clickButton( 'Continue' ),
+			] );
+		} );
+
+		it( 'Pick a design', async function () {
+			await freeSignupFlow.pickDesign( 'Upsidedown' );
+		} );
+
+		it( 'Land in Launchpad', async function () {
+			// Processing steps often takes longer than the default timeout.
+			await page.waitForURL( /launchpad/, { timeout: 20 * 1000 } );
+		} );
+
+		it( 'Launch site', async function () {
+			await freeSignupFlow.clickButton( 'Launch your site' );
+		} );
+
+		it( 'Navigated to Home', async function () {
+			await page.waitForURL( /home/ );
+
+			// How we handle modals within @automattic/calypso-e2e needs a rethink.
+			// Placing it as a "raw" selector for now.
+			await page.getByRole( 'dialog' ).getByText( 'Congrats, your site is live!' ).waitFor();
+		} );
+	} );
+
+	describe( 'Validate WordPress.com Free functionality', function () {
+		let sidebarComponent: SidebarComponent;
+
+		it( 'User has the selected free domain', async function () {
+			const urlRegex = `${ testUser.siteName }.wordpress.com`;
```

Comment:
I renamed it to `expectedURL` in 5a19b39a963f34c2d38b3b41f5f5e1c7170e6d24. 

Also renamed the test step because the user didn't explicitly select a domain name per sé.